packages/amplication-client/src/Resource/create-resource/CreateServiceWizard.tsx

```diff
@@ -34,36 +40,41 @@ type TData = {
   createResourceWithEntities: models.Resource;
 };
 
-const CreateServiceWizard: React.FC<Props> = ({ moduleClass, match }) => {
-  const { workspace, project } = match.params;
+const CreateServiceWizard: React.FC<Props> = ({ moduleClass }) => {
+  const { currentProject, onNewResourceCompleted } = useContext(AppContext);
+
   const serviceSettingsFields: MutableRefObject<serviceSettings> = useRef(
     serviceSettingsFieldsInitValues
   );
 
   const { trackEvent } = useTracking();
-  const history = useHistory();
-  const [createResourceWithEntities, { loading, data, error }] = useMutation<
-    TData
-  >(CREATE_RESOURCE_WITH_ENTITIES, {
-    update(cache, { data }) {
-      if (!data) return;
-
-      const queryData = cache.readQuery<{ resources: Array<models.Resource> }>({
-        query: GET_RESOURCES,
-      });
-      if (queryData === null) {
-        return;
-      }
-      cache.writeQuery({
-        query: GET_RESOURCES,
-        data: {
-          resources: queryData.resources.concat([
-            data.createResourceWithEntities,
-          ]),
-        },
-      });
-    },
-  });
+  const [createResourceWithEntities, { loading, error }] = useMutation<TData>(
+    CREATE_RESOURCE_WITH_ENTITIES,
+    {
+      onCompleted: (data) => {
+        onNewResourceCompleted(data.createResourceWithEntities);
+      },
+      update(cache, { data }) {
```

Comment:
why not use refetch instead of absolute update

Reply:
removed, we don't need it at this point